src/test/kotlin/com/terraformation/backend/tracking/ObservationServiceTest.kt

```diff
@@ -619,8 +621,35 @@ class ObservationServiceTest : DatabaseTest(), RunsAsUser {
     }
 
     @Test
-    fun `throws exception rescheduling an observation if the observation is not currently in Overdue state`() {
-      val observationId = insertObservation(state = ObservationState.Upcoming)
+    fun `throws exception rescheduling an observation if the observation is Completed`() {
+      val observationId =
+          insertObservation(state = ObservationState.Completed, completedTime = Instant.EPOCH)
+
+      val startDate = LocalDate.EPOCH
+      val endDate = startDate.plusDays(1)
+
+      assertThrows<ObservationRescheduleStateException> {
+        service.rescheduleObservation(observationId, startDate, endDate)
+      }
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+        strings =
+            [
+                "InProgress",
+                "Overdue",
+            ])
+    fun `throws exception rescheduling an In-Progress or Overdue observation if there are observed plots`(
+        stateName: String
+    ) {
+      insertPlantingSite()
+      insertPlantingZone(numPermanentClusters = 1, numTemporaryPlots = 1)
+      insertPlantingSubzone()
+      insertMonitoringPlot()
+
+      val observationId = insertObservation(state = ObservationState.valueOf(stateName))
+      insertObservationPlot(claimedBy = user.userId, completedBy = user.userId)
```

Comment:
Setting both claimedBy and completedBy here.
Was having trouble setting claimedBy to the value of completedBy in insertObservationPlot without breaking the alphabetical order of function arguments.

<img width="860" alt="Screen Shot 2023-10-31 at 4 23 42 PM" src="https://github.com/terraware/terraware-server/assets/1865174/ad341250-9dea-4cca-907a-b74d0ba5a254">

Reply:
Yeah, you'd have to change the order to make this work. What you're doing here is fine for now; we can tweak the insert function in a separate PR since we'd also want to update all the call sites which would be out of scope for this PR.